ipp/src/value.rs

```diff
@@ -326,14 +342,27 @@ impl<'a> Iterator for IppValueIterator<'a> {
 
     fn next(&mut self) -> Option<Self::Item> {
         match self.value {
-            IppValue::Array(ref array) | IppValue::Collection(ref array) => {
+            IppValue::Array(ref array) => {
                 if self.index < array.len() {
                     self.index += 1;
                     Some(&array[self.index - 1])
                 } else {
                     None
                 }
-            }
+            },
+            IppValue::Collection(ref map) => {
+                let array = Vec::from_iter(map.keys().cloned());
```

Comment:
With ordered collection it can be `map.iter().nth(index).map(v => v.1.clone())`

Reply:
I was getting ` expected `&IppValue`, found enum `IppValue` so I had to change it to `map.iter().nth(self.index).map(|v| &v.1.clone())` but still I got into `temporary value created here`
I provided alternative approach.